src/test/ui/consts/control-flow/feature-gate-const-if-match.rs

```diff
@@ -0,0 +1,27 @@
+// revisions: stock if_match
+
+#![feature(rustc_attrs)]
+#![cfg_attr(if_match, feature(const_if_match))]
+
+const _: i32 = if true { //[stock]~ ERROR `if` is not allowed in a `const`
+    5
+} else {
+    6
+};
+
+const _: i32 = match 1 { //[stock]~ ERROR `match` is not allowed in a `const`
+    2 => 3,
+    4 => 5,
+    _ => 0,
+};
+
+const fn foo() -> i32 {
+    if true { 5 } else { 6 } //[stock]~ ERROR `if` is not allowed in a `const fn`
+}
+
+const fn bar() -> i32 {
+    match 0 { 1 => 2, _ => 0 } //[stock]~ ERROR `match` is not allowed in a `const fn`
+}
+
+#[rustc_error]
```

Comment:
Would prefer splitting the file and having one be `// check-pass` so that stabilization becomes easier later on.

Reply:
I prefer the `// revisions:` approach to having two copies of the file while the feature flag exists.

Reply:
@ecstatic-morse In that case, please leave a note so that the file isn't removed whole-sale when we want to stabilize and so that it is converted into `// check-pass`.

Reply:
There's a comment at the top of the file now.